pkgs/tools/graphics/pngout/default.nix

```diff
@@ -1,34 +1,49 @@
-{lib, stdenv, fetchurl}:
+{ lib
+, stdenv
+, fetchurl
+, unzip
+}:
 
 let
-  folder = if stdenv.hostPlatform.system == "i686-linux" then "i686"
-  else if stdenv.hostPlatform.system == "x86_64-linux" then "x86_64"
-  else throw "Unsupported system: ${stdenv.hostPlatform.system}";
+  platforms = {
+    aarch64-linux = { folder = "aarch64"; ld-linux = "ld-linux-aarch64.so.1"; };
+    armv7l-linux = { folder = "armv7"; ld-linux = "ld-linux-armhf.so.3"; };
+    i686-linux = { folder = "i686"; ld-linux = "ld-linux.so.2"; };
+    x86_64-darwin = { folder = "."; };
+    x86_64-linux = { folder = "amd64"; ld-linux = "ld-linux-x86-64.so.2"; };
+  };
+  platform = platforms."${stdenv.hostPlatform.system}"
+    or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
```

Comment:
Those two things can be combined into one.

Reply:
Looks like the PR’s already been merged, but `platforms` is also used later in the definition of `meta.platforms`.